lib/std/zig/system/NativeTargetInfo.zig

```diff
@@ -355,55 +355,35 @@ fn detectAbiAndDynamicLinker(
         return result;
     }
 
-    const env_file = std.fs.openFileAbsoluteZ("/usr/bin/env", .{}) catch |err| switch (err) {
-        error.NoSpaceLeft => unreachable,
-        error.NameTooLong => unreachable,
-        error.PathAlreadyExists => unreachable,
-        error.SharingViolation => unreachable,
-        error.InvalidUtf8 => unreachable,
-        error.BadPathName => unreachable,
-        error.PipeBusy => unreachable,
-        error.FileLocksNotSupported => unreachable,
-        error.WouldBlock => unreachable,
-        error.FileBusy => unreachable, // opened without write permissions
-
-        error.IsDir,
-        error.NotDir,
-        error.InvalidHandle,
-        error.AccessDenied,
-        error.NoDevice,
-        error.FileNotFound,
-        error.FileTooBig,
-        error.Unexpected,
-        => return defaultAbiAndDynamicLinker(cpu, os, cross_target),
-
-        else => |e| return e,
-    };
-    defer env_file.close();
-
     // If Zig is statically linked, such as via distributed binary static builds, the above
-    // trick won't work. The next thing we fall back to is the same thing, but for /usr/bin/env.
-    // Since that path is hard-coded into the shebang line of many portable scripts, it's a
-    // reasonably reliable path to check for.
-    return abiAndDynamicLinkerFromFile(env_file, cpu, os, ld_info_list, cross_target) catch |err| switch (err) {
-        error.FileSystem,
-        error.SystemResources,
-        error.SymLinkLoop,
-        error.ProcessFdQuotaExceeded,
-        error.SystemFdQuotaExceeded,
-        => |e| return e,
-
-        error.UnableToReadElfFile,
-        error.InvalidElfClass,
-        error.InvalidElfVersion,
-        error.InvalidElfEndian,
-        error.InvalidElfFile,
-        error.InvalidElfMagic,
-        error.Unexpected,
-        error.UnexpectedEndOfFile,
-        error.NameTooLong,
-        // Finally, we fall back on the standard path.
-        => defaultAbiAndDynamicLinker(cpu, os, cross_target),
+    // trick won't work. The next thing we fall back to is the same thing, but for /usr/bin/env and /bin/bash.
+    // Since these paths are hard-coded into the shebang line of many portable scripts, they are a
+    // reasonably reliable paths to check for.
+    return result: {
+        break :result abiAndDynamicLinkerFromPath("/usr/bin/env", cpu, os, ld_info_list, cross_target) catch {
+            // If we can't parse /usr/bin/env ELF file (for example, if coreutils was builded with --enable-single-binary and /usr/bin/env is actually a script), try /bin/bash
```

Comment:
For example, Gentoo's `sys-apps/coreutils` with USE flag `multicall`, or Red Hat Universal Base Image 9's `coreutils-single` https://github.com/ziglang/zig/issues/12156#issuecomment-1187036721